### PR TITLE
distant: init at 0.20.0

### PR DESCRIPTION
**Pull request by nyxar77**
add distant remote development tool, used in also in libraries like distant.nvim

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/tools/networking/distant/default.nix
Comment (mweinelt) [inline pkgs/tools/networking/distant/default.nix:1]: Move to pkgs/by-name/di/distant/package.nix

File: pkgs/tools/networking/distant/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  doCheck = false;
+
+  meta = {
+    description = "Distant - A fast, secure remote development tool";
+    homepage = "https://github.com/chipsenkbeil/distant";
+    license = lib.licenses.mit;
+    maintainers = [
+      "chipsenkbeil"
+      "westernwontons"
+    ];
```
Comment (mweinelt) [inline pkgs/tools/networking/distant/default.nix:36]: Maintainers are supposed to be handles defined in lib/maintainers.nix.

Check the header of that file and create a separate commit for each maintainer. The package commit should come after.

**Review by mweinelt — CHANGES_REQUESTED** [review]
The commit message is not following the contribution guidelines.

https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#commit-conventions

File: pkgs/tools/networking/distant/default.nix
Comment (fe2-Nyxar) [reply pkgs/tools/networking/distant/default.nix:1]: first time, thanks for your help

File: pkgs/tools/networking/distant/default.nix (hunk 1 above)
Comment (fe2-Nyxar) [reply pkgs/tools/networking/distant/default.nix:36]: i couldn't find the file you are talking about, also these are the maintainers of the project so logically they can be here

Comment (mweinelt) [reply pkgs/tools/networking/distant/default.nix:36]: The file is at `./maintainers/maintainer-list.nix`, but the references are available at `lib.maintainers`.

Comment (fe2-Nyxar) [reply pkgs/tools/networking/distant/default.nix:36]: so it should be like add maintainers to the maintainers/maintainer-list.nix
and add this line in default.nix?
```nix
maintainers = [ lib.maintainers.user ]
```
tell me if i'm corret cause i'm confused

Comment (mweinelt) [reply pkgs/tools/networking/distant/default.nix:36]: Yes, add a maintainer object at the correct location to the file and reference it as `lib.maintainters.chipseenkbeil` if you name the object that.

Comment (fe2-Nyxar) [reply pkgs/tools/networking/distant/default.nix]: thank you so much, big help
i just got done, tell me if i'm missing anything :)

**Comment by fe2-Nyxar** [comment]
@ofborg eval

File: maintainers/maintainer-list.nix [hunk 2]
```diff
@@ -3894,6 +3894,12 @@
     githubId = 631802;
     keys = [ { fingerprint = "099E 3F97 FA08 3D47 8C75  EBEC E0EB AD78 F019 0BD9"; } ];
   };
+  chipsenkbeil = {
+    email = "nyxar49@gmail.com";
```
Comment (winterqt) [inline maintainers/maintainer-list.nix:3898]: Why are these all your email?

Also: these seem to be the maintainers of the upstream project. You should only add yourself as a maintainer in this case.

**Review by winterqt — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix (hunk 2 above)
Comment (fe2-Nyxar) [reply maintainers/maintainer-list.nix]: thanks for clarification, i've removed them

File: pkgs/by-name/di/distant/package.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  openssl,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "distant";
+  version = "0.20.0";
+
+  src = fetchFromGitHub {
+    owner = "chipsenkbeil";
+    repo = pname;
```
Comment (pluiedev) [inline pkgs/by-name/di/distant/package.nix:16]: `repo = pname` is an anti-pattern
```suggestion
    repo = "distant";
```

File: pkgs/by-name/di/distant/package.nix [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    rev = "v${version}";
```
Comment (pluiedev) [inline pkgs/by-name/di/distant/package.nix:17]:
```suggestion
    rev = "refs/tags/v${version}";
```

File: pkgs/by-name/di/distant/package.nix [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    hash = "sha256-DcnleJUAeYg3GSLZljC3gO9ihiFz04dzT/ddMnypr48=";
+  };
+
+  cargoHash = "sha256-7MNNdm4b9u5YNX04nBtKcrw+phUlpzIXo0tJVfcgb40=";
+  nativeBuildInputs = [
+    pkg-config
+    perl
+  ];
+  buildInputs = [ openssl ];
+  doCheck = false;
```
Comment (pluiedev) [inline pkgs/by-name/di/distant/package.nix:27]: Add a comment explaining why tests are disabled (e.g. no tests/failing tests)

File: pkgs/by-name/di/distant/package.nix [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+  meta = {
+    description = "Distant - A fast, secure remote development tool";
```
Comment (pluiedev) [inline pkgs/by-name/di/distant/package.nix:30]:
```suggestion
    description = "Fast, secure remote development tool";
```

**Review by pluiedev — CHANGES_REQUESTED** [review]
Some nitpicks. Also, please squash your commits into just two:
* `maintainers: add fe2-Nyxar`
* `distant: init at 0.20.0`

File: pkgs/by-name/di/distant/package.nix (hunk 4 above)
Comment (fe2-Nyxar) [reply pkgs/by-name/di/distant/package.nix:17]: I can't squash right now because I have some requested changes

**Pull request closed without merging**